File: src/EChamado/Client/EChamado.Client/Services/RoleService.cs
using EChamado.Client.Models;
using EChamado.Shared.Responses;
using System.Net.Http.Json;

namespace EChamado.Client.Services;

public class RoleService
{
    private readonly HttpClient _httpClient;

    public RoleService(HttpClient httpClient)
    {
        _httpClient = httpClient;
    }

    /// <summary>
    /// Busca todos os perfis
    /// </summary>
    public async Task<List<RoleResponse>> GetAllAsync(int pageSize = 100)
    {
        var result = await _httpClient.GetFromJsonAsync<BaseResultList<RoleResponse>>($"v1/role?PageSize={pageSize}");
        return result?.Data?.ToList() ?? new List<RoleResponse>();
    }

    /// <summary>
    /// Busca um perfil por ID
    /// </summary>
    public async Task<RoleResponse?> GetByIdAsync(Guid id)
    {
        try
        {
            var result = await _httpClient.GetFromJsonAsync<BaseResult<RoleResponse>>($"v1/role/{id}");
            return result?.Data;
        }
        catch (HttpRequestException)
        {
            return null;
        }
    }

    /// <summary>
    /// Busca um perfil por nome
    /// </summary>
    public async Task<RoleResponse?> GetByNameAsync(string name)
    {
        try
        {
            var result = await _httpClient.GetFromJsonAsync<BaseResult<RoleResponse>>($"v1/role/name/{name}");
            return result?.Data;
        }
        catch (HttpRequestException)
        {
            return null;
        }
    }

    /// <summary>
    /// Cria um novo perfil
    /// </summary>
    public async Task<Guid> CreateAsync(CreateRoleRequest request)
    {
        var response = await _httpClient.PostAsJsonAsync("v1/role", request);
        response.EnsureSuccessStatusCode();
        var result = await response.Content.ReadFromJsonAsync<BaseResult<Guid>>();
        return result?.Data ?? Guid.Empty;
    }

    /// <summary>
    /// Atualiza um perfil existente
    /// </summary>
    public async Task UpdateAsync(Guid id, UpdateRoleRequest request)
    {
        var response = await _httpClient.PutAsJsonAsync($"v1/role/{id}", request);
        response.EnsureSuccessStatusCode();
    }

    /// <summary>
    /// Deleta um perfil
    /// </summary>
    public async Task DeleteAsync(Guid id)
    {
        var response = await _httpClient.DeleteAsync($"v1/role/{id}");
        response.EnsureSuccessStatusCode();
    }
}

File: src/Frontend/nextjs/src/services/role.service.ts
import apiClient from '@/lib/api-client';
import { BaseResult, BaseResultList, RoleResponse, CreateRoleRequest, UpdateRoleRequest } from '@/types/api';

export const roleService = {
  async getAll(pageSize = 100) {
    const { data } = await apiClient.get<BaseResultList<RoleResponse>>('/v1/role', { params: { PageSize: pageSize } });
    return data;
  },
  async getById(id: string) {
    const { data } = await apiClient.get<BaseResult<RoleResponse>>(`/v1/role/${id}`);
    return data;
  },
  async getByName(name: string) {
    const { data } = await apiClient.get<BaseResult<RoleResponse>>(`/v1/role/name/${name}`);
    return data;
  },
  async create(req: CreateRoleRequest) {
    const { data } = await apiClient.post<BaseResult<string>>('/v1/role', req);
    return data;
  },
  async update(id: string, req: UpdateRoleRequest) {
    const { data } = await apiClient.put<BaseResult>(`/v1/role/${id}`, req);
    return data;
  },
  async delete(id: string) {
    const { data } = await apiClient.delete<BaseResult>(`/v1/role/${id}`);
    return data;
  },
};
